arb_os/decompression.mini

```diff
@@ -223,6 +235,74 @@ public impure func decompressAndVerifyEcdsaSignedTx(
     );
 }
 
+public impure func decompressAndVerifyBLSSignedTxBatch(
+    ba: ByteArray,
+    originalRequest: IncomingRequest,
+    queueToFill: Queue,
+) -> option<Queue> {
+    let stream = bytestream_skipBytes(bytestream_new(ba), 1)?;
+
+    let (stream, numMsgs) = rlp_decodeUint(stream)?;
+    let msgHashes = newarray<BLSHashedMessage>(numMsgs);
+    let signerKeys = newarray<BLSKey>(numMsgs);
+    let txs = newarray<TxRequestData>(numMsgs);
+
+    let (stream, sigx) = bytestream_get256(stream)?;
+    let (stream, sigy) = bytestream_get256(stream)?;
+    let aggregatedSig = struct { x: sigx, y: sigy };
+
+    let i = 0;
+    let domain = bytearray_set256(bytearray_new(0), 0, 0);
+    while(i < numMsgs) {
+        // the signature on this tx covers all of the data associated with this tx
+        // we remember the start of the stream, so we can extract the full data later, once we know where the end is
+        let startStream = stream;
+
+        let (us, signerAddr) = decompressAddress(stream)?;
+        stream = us;
+
+        let (us, txSize) = rlp_decodeUint(stream)?;
+        stream = us;
+
+        let (us, txData) = bytestream_getN(stream, txSize)?;
+        stream = us;
+
+        let fullDataSize = bytestream_bytesReadSoFar(stream)-bytestream_bytesReadSoFar(startStream);
+        let (_, fullData) = bytestream_getN(startStream, fullDataSize)?;
+        msgHashes = msgHashes with { [i] = bls_hashToPoint(domain, fullData)? };
+
+        signerKeys = signerKeys with {
+            [i] = account_getBlsKey(accountStore_get(getGlobalAccountStore(), signerAddr))?
+        };
+
+        let aggregator = originalRequest.sender;
+        if let Some(aggInfo) = originalRequest.aggregator {
+            aggregator = aggInfo.aggregator;
+        }
+
+        let (_, newTx) = decompressTx(bytestream_new(txData), Some(signerAddr), None<FunctionTable>)?;
+        newTx = newTx with {
+            incomingRequest: originalRequest with {
+                requestId: uint(hash(bytes32(originalRequest.requestId), bytes32(i)))
```

Comment:
BLS transactions should have their request ID be calculable before they enter into the inbox like ECDSA transactions

Reply:
Changed this to have the request ID be the keccak256 of the data that the user signed.